release/models/interfaces/openconfig-interfaces.yang

```diff
@@ -1048,10 +1055,56 @@ module openconfig-interfaces {
 
     leaf carrier-transitions {
       type oc-yang:counter64;
+      status deprecated;
       description
         "Number of times the interface state has transitioned
         between up and down since the time the device restarted
-        or the last-clear time, whichever is most recent.";
+        or the last-clear time, whichever is most recent.
+
+        Please use interface-transitions instead, which has
+        similar, but more precisely specified, semantics and a
+        clearer name.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf interface-transitions {
+      type oc-yang:counter64;
+      description
+        "The total number of times the interface state (oper-status)
+        has either transitioned to 'UP' state from any other state, or
+        from state 'UP' to any other state.  I.e., an interface flap
+        from UP to DOWN back to UP increments the counter by 2.
+        Transitions between any other interface states other than to
+        or from 'UP' state are not included in the counter.
+
+        Discontinuities in the value of this counter can occur
+        when the device is re-initialization as indicated by the
+        value of 'last-clear'.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf link-transitions {
+      type oc-yang:counter64;
+      description
+        "This is the number of times that the underlying link state
+        (e.g., at the optical receiver) has transitioned to or from
+        'UP' state before any holdtime, dampening, or other processing
+        has been applied that could suppress an update to the interface
+        'oper-status' and corresponding interface-transitions counter.
+
+        The counter is incremented both when the link transitions
+        to 'UP' state from any other link state and also when the link
+        transitions from 'UP' state to any other link state, i.e., an
+        interface flap from UP to DOWN back to UP increments the
+        counter by 2.
+
+        Implementations are not required to count all transitions,
```

Comment:
Did Jeff's point about this possibly also apply to interface-transitions? He brought up some point (I think) about the system coming up or maybe a linecard coming up?

Reply:
I don't think that it should.

Link transitions could happen at high frequency and there are good reasons that such events might get lost or be suppressed and not necessarily truly represent the number of transitions.

But I think that interface-transitions is different, this counter should only be changed when the interface-state field is changed, which I think systems should be able to properly sequence.  Otherwise it feels like it would be getting too much into implementation details.

Reply:
Agreed, all interface transitions are expected to be set and observed by the overall device management and control software.